builds/misc/templates/mqtt-perf-test-case.yaml

```diff
@@ -0,0 +1,85 @@
+  type: string
+  default: false
+- name: id # current test run id; required
+  type: string
+  default: false
+- name: label # test run label; required
+  type: string
+  default: false
+- name: publishers # number of publisher clients; required
+  type: number
+  default: false
+- name: subscribers # number of subscriber clients; required
+  type: number
+  default: false
+- name: topics # number of unique topics to use for the test case; required
+  type: number
+  default: false
+- name: payload # a message payload size in bytes; required
+  type: number
+  default: false
+- name: qos # qos of messages; required
+  type: number
+  default: false
+
+jobs:
+
+- job: "run_pub_${{ parameters['id'] }}"
+  pool:
+    name: $(pool.name)
+    demands: mqtt-perf-tests-pub
+  displayName: Test Case Pub ${{ parameters['publishers'] }}p / ${{ parameters['subscribers'] }}s / ${{ parameters['topics'] }}t / ${{ parameters['payload'] }}b / QoS${{ parameters['qos'] }}
+  dependsOn: ${{ parameters['parentJob'] }}
+  steps:
+  - checkout: none
+  - script: |
+      $(benchmarkToolPath)/mqtt-benchmark -pub -broker $(brokerEndpoint) -runId $(runId) -caseId ${{ parameters['id'] }} -clients ${{ parameters['publishers'] }} -duration $(test.duration) -size ${{ parameters['payload'] }} -topics ${{ parameters['topics'] }} -qos ${{ parameters['qos'] }} -dop $(clientDop) -waitFor $(subscriber.readyEndpoint) -quiet -panic
+    displayName: 'Running test case'
+    env:
+      LOGANALYTICS_CUSTOMER_ID: $(logs.workspaceid)
+      LOGANALYTICS_SHARED_KEY: $(logs.sharedkey)
+      LOGANALYTICS_LOG_NAME: $(logs.logname)
+    continueOnError: true 
+
+- job: "run_sub_${{ parameters['id'] }}"
+  pool:
+    name: $(pool.name)
+    demands: mqtt-perf-tests-sub
+  displayName: Test Case Sub ${{ parameters['publishers'] }}p / ${{ parameters['subscribers'] }}s / ${{ parameters['topics'] }}t / ${{ parameters['payload'] }}b / QoS${{ parameters['qos'] }}
+  dependsOn: ${{ parameters['parentJob'] }}
+  steps:
+  - checkout: none
+  - script: |
+      $(benchmarkToolPath)/mqtt-benchmark -sub -broker $(brokerEndpoint) -runId $(runId) -caseId ${{ parameters['id'] }} -clients ${{ parameters['subscribers'] }} -duration $(test.duration) -qos ${{ parameters['qos'] }} -dop $(clientDop) -quiet -panic
+    displayName: 'Running test case'
+    env:
+      LOGANALYTICS_CUSTOMER_ID: $(logs.workspaceid)
+      LOGANALYTICS_SHARED_KEY: $(logs.sharedkey)
+      LOGANALYTICS_LOG_NAME: $(logs.logname)
+    continueOnError: true 
+
+- job: "run_completed_${{ parameters['id'] }}"
+  pool:
+    name: $(pool.name)
+    demands: mqtt-perf-tests-sub
+  displayName: Cleanup
+  dependsOn:   
+  - "run_sub_${{ parameters['id'] }}"
+  - "run_pub_${{ parameters['id'] }}"
+  steps:
+  - checkout: none
+  - task: SSH@0
+    displayName: Stop broker AMD64
+    inputs:
+      sshEndpoint: ${{ variables.brokerSsh }} 
+      runOptions: commands
+      commands: |
+        docker restart mqtt-broker
```

Comment:
display name is "Stop broker", but why use "docker restart"?

Reply:
Thanks, it should be restart. Fixed the display name

Reply:
should it be better to remove the container? to make sure there should not have any data left from previous test.

Reply:
please see comment below